components/recurring-contributions/UpdatePaymentMethodPopUp.js

```diff
@@ -0,0 +1,280 @@
+import React, { Fragment, useState } from 'react';
+import PropTypes from 'prop-types';
+import { graphql } from '@apollo/react-hoc';
+import { useMutation } from '@apollo/react-hooks';
+import { PlusCircle } from '@styled-icons/boxicons-regular/PlusCircle';
+import { MoneyCheck } from '@styled-icons/fa-solid/MoneyCheck';
+import themeGet from '@styled-system/theme-get';
+import gql from 'graphql-tag';
+import { get, uniqBy } from 'lodash';
+import { withRouter } from 'next/router';
+import { defineMessages, FormattedDate, FormattedMessage, useIntl } from 'react-intl';
+import styled from 'styled-components';
+
+import { formatCurrency } from '../../lib/currency-utils';
+import { getErrorFromGraphqlException } from '../../lib/errors';
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import { getPaymentMethodName, paymentMethodExpiration } from '../../lib/payment_method_label';
+
+import { Box, Flex } from '../Grid';
+import CreditCard from '../icons/CreditCard';
+import GiftCard from '../icons/GiftCard';
+import StyledButton from '../StyledButton';
+import StyledHr from '../StyledHr';
+import StyledRadioList from '../StyledRadioList';
+import { P } from '../Text';
+
+const PaymentMethodBox = styled(Flex)`
+  border-top: 1px solid ${themeGet('colors.black.300')};
+`;
+
+const messages = defineMessages({
+  cancel: {
+    id: 'actions.cancel',
+    defaultMessage: 'Cancel',
+  },
+  update: {
+    id: 'subscription.updateAmount.update.btn',
+    defaultMessage: 'Update',
+  },
+  updatePaymentMethod: {
+    id: 'subscription.menu.editPaymentMethod',
+    defaultMessage: 'Update payment method',
+  },
+});
+
+const updatePaymentMethodMutation = gqlV2/* GraphQL */ `
+  mutation updatePaymentMethod($order: OrderReferenceInput!, $paymentMethod: PaymentMethodReferenceInput!) {
+    updateOrder(order: $order, paymentMethod: $paymentMethod) {
+      id
+    }
+  }
+`;
+
+const UpdatePaymentMethodPopUp = ({ data, setMenuState, contribution, createNotification, setShowPopup }) => {
+  const intl = useIntl();
+  const [selectedPaymentMethod, setSelectedPaymentMethod] = useState(null);
+  const [submitUpdatePaymentMethod, { loadingUpdatePaymentMethod }] = useMutation(updatePaymentMethodMutation, {
+    context: API_V2_CONTEXT,
+  });
+
+  // data handling\
+  const minBalance = 50; // Minimum usable balance for virtual card
+
+  const getPaymentMethodIcon = pm => {
```

Comment:
This and `getPaymentMethodMetadata` are the same than the ones in https://github.com/opencollective/opencollective-frontend/blob/3253007809f6e32250232f5763a3cde3c28d9a38/components/contribution-flow/StepPayment.js. Can we refactor to avoid code duplication?